internal/core/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+add_library(vdscore
```

Comment:
Because it builds the c++ portion of the go `core` package 😄 And it's not in anyway intended to be public target so it doesn't need a "unique" or "stand-out" name

Reply:
I honestly hate the word "core" that we use everywhere. But if you need to name a meaningless thing, general meaningless name works fine I guess.
Now we would have core library inside core go package.Very helpful and easy to distinguish.

Could we maybe rename to cppcore/cppcoretests instead? Would that be ok with you?

Reply:
> Now we would have core library inside core go package.
This is exactly my point, but if you don't like it, leave it